lib/ansible/modules/user.py

```diff
@@ -674,7 +676,22 @@ def create_user_useradd(self):
 
         cmd.append(self.name)
         (rc, err, out) = self.execute_command(cmd)
-        if not self.local or rc != 0 or self.groups is None or len(self.groups) == 0:
+        if not self.local or rc != 0:
+            return (rc, err, out)
+
+        if self.expires is not None:
+            if self.expires < time.gmtime(0):
+                lexpires = -1
+            else:
+                # Convert seconds since Epoch to days since Epoch
+                lexpires = int(math.floor(float(self.module.params['expires']))) // 86400
```

Comment:
I'm not sure the `float` conversion is necessary. The `expires` parameter is specified as `type='float'` which means it has already gone through `check_type_float()` which means it has already been converted to a `float`.

I had to look into why `int(math.floor())` was necessary. `math.floor()` returns a `float` on Python 2 and an `int` in Python 3. I learned a bit from this line, so thanks for that!

Reply:
I learned as well :-). I stumbled across this as the tests failed on some platforms before https://github.com/ansible/ansible/pull/72022/commits/1695b5ed8b33afee8fba34ae759eaaeb98abc15b and the difference was as you state that the failing platforms ran the tests on Python 2 while the succeeding ones ran on Python 3. Hence the great wealth of platforms tested was very helpful in detecting this issue.
Done in afd6e08.